website/content/docs/configuration/client.mdx

```diff
@@ -355,6 +355,100 @@ see the [drivers documentation](/docs/drivers).
   files on the client host via the `file` function. By default templates can
   access files only within the [task working directory].
 
+- `max_stale` `(string: "5s")` - # This is the maximum interval to allow "stale"
+  data. By default, only the Consul leader will respond to queries. Requests
+  to a follower will forward to the leader. In large clusters with many requests,
+  this is not as scalable, so this option allows any follower to respond to a query,
+  so long as the last-replicated data is within these bounds. Higher values result
+  in less cluster load, but are more likely to have outdated data.
+
+- `wait` `(Code: nil)` - Defines the minimum and maximum amount of time to wait
+  for the cluster to reach a consistent state before rendering a template. This
+  is useful to enable in systems that are experiencing a lot of flapping, because
+  it will reduce the number of times a template is rendered. This configuration is
+  also exposed at the _task level_ to allow overrides per task.
+
+  ```hcl
+  wait {
+    enabled = true
+    min     = "5s"
+    max     = "10s"
+  }
+  ```
+
+- `wait_bounds` `(Code: nil)` - Defines client level lower and upper bounds for
+  per-template `wait` configuration. If the individual template configuration has
+  a `min` lower than `wait_bounds.min` or a `max` greater than the `wait_bounds.max`,
+  the bounds will be enforced, and the template `wait` will be adjusted before being
+  sent to `consul-template`.
+
+  ```hcl
+  wait_bounds {
+    enabled = true
+    min     = "5s"
+    max     = "10s"
+  }
+    ```
+
+- `block_query_wait` `(string: "60s")` - This is amount of time in seconds to wait
+  on a blocking query for. Many endpoints in Consul support a feature known as
+  "blocking queries". A blocking query is used to wait for a potential change
+  using long polling.
+
+- `consul_retry` `(Code: nil)` - This controls the retry behavior when an error is
+  returned from Consul. Consul Template is highly fault tolerant, meaning it does
+  not exit in the face of failure. Instead, it uses exponential back-off and retry
+  functions to wait for the cluster to become available, as is customary in distributed
+  systems.
+
+  ```hcl
+  consul_retry {
+    # This enables retries.
+    enabled = true
+    # This specifies the number of attempts to make before giving up. Each
+    # attempt adds the exponential backoff sleep time. Setting this to
+    # zero will implement an unlimited number of retries.
+    attempts = 12
```

Comment:
These are not default values, but rather example values. The Vault & Consul retry default configuration in `consul-template` are empty structs with no fields set. We don't have to provide that, because `consul-template` has a `Finalize` method that handles defaulting the values if callers don't set them.

Reply:
I'm going to resolve this conversation for now, but please re-open if I haven't addressed all the concerns here.

Reply:
LGTM